src/main/java/org/broadinstitute/hellbender/metrics/QualityYieldMetrics.java

```diff
@@ -0,0 +1,143 @@
+    public long Q20_BASES;
+
+    /** The number of bases in PF reads that achieve quality score 20 or higher */
+    public long PF_Q20_BASES;
+
+    /** The number of bases in all reads that achieve quality score 30 or higher */
+    public long Q30_BASES;
+
+    /** The number of bases in PF reads that achieve quality score 30 or higher */
+    public long PF_Q30_BASES;
+
+    /** The sum of quality scores of all bases divided by 20 */
+    public long Q20_EQUIVALENT_YIELD;
+
+    /** The sum of quality scores of all bases divided by 20 */
+    public long PF_Q20_EQUIVALENT_YIELD;
+
+    //adds stats for the read
+    public QualityYieldMetrics addRead(final GATKRead read){
+        TOTAL_READS++;
+        final int length = read.getLength();
+
+        final boolean isPfRead = !read.failsVendorQualityCheck();
+        if (isPfRead) {
+            PF_READS++;
+            PF_BASES += length;
+        }
+
+        TOTAL_BASES += length;
+
+        final byte[] quals;
+        if (useOriginalQualities) {
+            byte[] tmp = ReadUtils.getOriginalBaseQualities(read);
+            if (tmp == null) {
+                tmp = read.getBaseQualities();
+            }
+            quals = tmp;
+        } else {
+            quals = read.getBaseQualities();
+        }
+
+        // add up quals, and quals >= 20
+        for (final byte qual : quals) {
+            Q20_EQUIVALENT_YIELD += qual;
+            if (qual >= 20) {
+                Q20_BASES++;
+            }
+            if (qual >= 30) {
+                Q30_BASES++;
+            }
+
+            if (isPfRead) {
+                PF_Q20_EQUIVALENT_YIELD += qual;
+                if (qual >= 20) {
+                    PF_Q20_BASES++;
+                }
+                if (qual >= 30) {
+                    PF_Q30_BASES++;
+                }
+            }
+        }
+        return this;
+    }
+
+    //merges two objects
+    public QualityYieldMetrics merge(final QualityYieldMetrics that) {
+        Utils.nonNull(that);
+        if (useOriginalQualities != that.useOriginalQualities){
+            throw new IllegalArgumentException("must have the same value for useOriginalQualities");
+        }
+        if (finalized || that.finalized){
+            throw new IllegalArgumentException("can;t merge objects when the calculations are finalized");
+        }
+
+        TOTAL_READS  += that.TOTAL_READS;
+        PF_READS     += that.PF_READS;
+        READ_LENGTH  += that.READ_LENGTH;
+        TOTAL_BASES  += that.TOTAL_BASES;
+        PF_BASES     += that.PF_BASES;
+        Q20_BASES    += that.Q20_BASES;
+        PF_Q20_BASES += that.PF_Q20_BASES;
+        Q30_BASES    += that.Q30_BASES;
+        PF_Q30_BASES += that.PF_Q30_BASES;
+        Q20_EQUIVALENT_YIELD += that.Q20_EQUIVALENT_YIELD;
+        PF_Q20_EQUIVALENT_YIELD += that.PF_Q20_EQUIVALENT_YIELD;
+
+        return this;
+    }
+
+    //completes the calculations
+    public QualityYieldMetrics finish() {
```

Comment:
Right - that will be part of phase II, when we'll have more than reducible collector.